### ISSUE TITLE
New Name for the "Chats" folder

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-core#47]
Currently, we move messages sent by Delta Chat and replies to them to the special IMAP-folder "Chats". If this folder does not exist, it is created by Delta Chat (yes, there are [some bugs](https://github.com/deltachat/deltachat-android/issues/108) with that, but this is the plan how it should work :)

However, the name "Chats" was not a good choice as it is reserved by some providers, Google among them. 

So, currently, we're searching for a new generic name for this folder.

- the name should be unique
- it should be clear what gets in there
- it should not be reserved, so. eg. `Chat` may be a bad choice as well
- there is no need to bind the name to "Delta Chat", however, it is not impossible (maybe even the best choice) (we avoid this binding also for other generic things, eg. the `Chat:` prefix in the subject or the [internal header names](https://delta.chat/en/spec))  
- it is not planned to use customized names for the user (different names may make things even more complicated in the future)

Suggestions up to now:

`DeltaChats`

`DeltaChat`

`IMAPChat`

**Comment by testbird** [comment]
Hmm, compared to the other common default folders like Sent, Junk, Draft, Spam, Trash, ..., the content does not consist of formal E-Mail Letters...

So other ideas: Messages, Conversation, ...?

**Comment by testbird** [comment]
Messenger
delta.chat  <-- actually hinting to the URL if anybody is not in the know.

**Comment by r10s** [comment]
> the content does not consist of formal E-Mail Letters...

Well, replies from normal MUAs to chats started in Delta will be moved there, too. However, if you mean the typically missing "Dear madam/sir ... Best regards ..." - that's true.

**Comment by STPKITT** [comment]
Blank in the name are not allowed?

The rest of the decision is highly dependend on which kind of people you're targeting. Experts/Nerds etc. will always comprehend what that new folder in their inbox is all about no matter what name it has, BUT every other person not into tech will not.
So I highly recommend that "Delta" is part of the name, I personally would vote for DeltaChat.

**Comment by r10s** [comment]
> Blanks in the name are not allowed?

Not sure about what the spec says. But I'm quite sure, there will be at least one app or one agent having problem with spaces in foldernames. It may also be more difficult for the user, eg. when adding a filter or whatever. 
So, maybe it is a good idea not to use spaces, event if the standard says that we can use some and even if Delta has no problems with them.

**Comment by STPKITT** [comment]
Ok, I agree. No blanks often means less trouble.

**Comment by Hocceruser** [comment]
I'm in favour of two separate folders, DeltaChat-sent and DeltaChat-inbox.

**Comment by r10s** [comment]
> I'm in favour of two separate folders, DeltaChat-sent and DeltaChat-inbox.

But this would tear apart the threads ... something most users don't want.

**Comment by kollokollo** [comment]
In my opinion, one can leave it as it is (Folder is called "Chats", all deltachat messages go there, they are encrypted anyway, so users will barely look into it anyways.).

**Comment by r10s** [comment]
Well, if we leave it as it is, the folder does not work for some some providers, see above.

**Comment by Hocceruser** [comment]
r10s wrote:
> > I'm in favour of two separate folders, DeltaChat-sent and DeltaChat-inbox.
> 
> But this would tear apart the threads ... something most users don't want.

Personally, I found it rather a bit surprising that the messages I sent and those I received are in the same folder, because with "normal" email clients there are two separate folders. On the other hand, of course, you are right as well.

Anyway - If there is only one folder, I'd call it DeltaChat.

**Comment by judgy** [comment]
The folder shouldn't have any relation to 'Delta'. I would prefere ImapChat or EmailChat.
I'm checking if I can develop a Windows (WPF) version of Delta Chat, but I will not name it Delta-Chat for Windows, because maybe also other people will try develop a Windows version with other frameworks like QT. But I'll try to use the standard of Delta Chat (http://delta.chat/en/standards).
So the folder chat should be something common.

**Comment by testbird** [comment]
What was the problem with `Chat`, was reserved where for what?
 `Chats` not working with receipients hosted at data collectors could actually be a feature.

**Comment by kollokollo** [comment]
Maybe on can leeave it as it is (means: Folder is named "Chats") and only for thouse email-servers, where die Folder cannot be created use a different one as a fallback solution.

**Comment by r10s** [comment]
- "Chats" is reserved in Gmail, you cannot create an IMAP-folder with this name. Maybe this comes from the days where you could use XMPP via Google Talk.

- different Names on different Installation should be avoided (such things are known to make problems later ;), so, no fallback, but a functional default. I think `DeltaChat` is the way to go ...

**Comment by testbird** [comment]
More generic ideas to find a convention:
EmailChat
Conversations
Messenger

**Comment by mva1985** [comment]
imho if you go with one folder I'd call it DeltaChat as well...  kinda makes sense

**Comment by testbird** [comment]
Well this is an issue for deltachat-core, and it is written to support other email chat clients.

The "Email-chat specification" (see https://delta.chat/en/spec) currently defines a client independent default folder `Chats`. If this is needing to change, it would make sense for the Email-chat specification go for a generic folder name, if it really wants to become a standard.

**Comment by testbird** [comment]
@r10s have you thought about registering email.chat? (at least for the spec)

**Comment by r10s** [comment]
@testbird what do you mean with "registering"?

**Comment by testbird** [comment]
The DNS domain.

**Comment by r10s** [comment]
Ahh, okay. No I have not registered it - nor thought about it yet :)

**Comment by testbird** [comment]
All right, your're setting the standards here. :-)

**Comment by Hocceruser** [comment]
So what name did you choose, r10s? And after this problem is eliminated, will DC be on the Play store then?

**Comment by csb0730** [comment]
I see the 1st comment on this issue, but why not making an expert option with a configurable name?
This would solve the problem with reserved names forever.

**Comment by r10s** [comment]
Well known places and names make things much easier in practise, IMHO.

**Comment by Hocceruser** [comment]
I think that for most people (including me) the name of the chats folder is trivial. It should be obvious what is saved there but that's all. So you could just take any of the suggestions made here.

**Comment by testbird** [comment]
Here:  E-Chat

**Comment by testbird** [comment]
"E-chat" => standing for [Email-chat specification](https://delta.chat/en/spec)

**Comment by csb0730** [comment]
Simple proposal:

email = E-Mail
echat = E-Chat

Maybe "eChat" or "echat" or "ECHAT" could be a suitable name if a new one is necessary?

**Comment by hpk42** [comment]
How about "DeltaChat"?  If i installed an app called "delta chat" 
and find such an imap folder it's somewhat clear how it came there 
or what app relates to it ...

**Comment by testbird** [comment]
If you install them another chat app it would not be clear why the Email-chat specification folder is named after **another** (nevertheless the pioneering) app.

**Comment by testbird** [comment]
https://blog.fastmail.com/2016/07/13/whats-in-a-name-mailbox-names-via-imap/
Looks to me the standard folder should be subfolder, and let MUAs map them to the root folder if desired?
I could not find anything forbidding the "-". 
 `INBOX.E-Chat`

**Comment by Hocceruser** [comment]
It is 2 months since the last post on this issue now and I think that we should really decide now.

What about kind of a vote: everyone can do a suggestion by posting his/her favourite name. Voting is done by giving thumbs up / +1 to the name(s) you approve. (at the GitHub web frontend)

**Comment by Hocceruser** [comment]
Like this:
EmailChats

**Comment by r10s** [comment]
@Hocceruser yes, we can try a vote to get an idea, why not. however, might be that the chosen name won't work in the end.

regarding the time: the is strongly related to the more complicated issue with Google, Xoauth and so on. I think when this is started to being fixed, the name will be chosen.

**Comment by testbird** [comment]
I have no preference between EmailChat or EmailChats, so just put my two thumbs up. :-)

**Comment by t-oster** [comment]
At Yahoo, "Chats" also does not work. I prefer EmailChats

**Comment by r10s** [comment]
as @t-oster has fixed some other issues with respect to the "Chats" folder (see eg. #173), i think it is time to make a decision here. Summarizing the comment, we have two names:

- `DeltaChat`  
Advantage: easier to recognize for non-tech people  
Disadvantage: no [standard character](https://github.com/deltachat/spec/blob/master/spec.md#email-chat-specification) if 3rd parties want to create a compatible chat app

- `EmailChats`
Advantage: standard-character  
Disadvantage: harder to recognize for non-tech people

So, the advantages of the one are the disadvantages of the other :)
So, maybe we should think about how hard the disadvantages are ...

**Comment by t-oster** [comment]
IMHO EmailChats is not harder to recognize for non tech people, so I vote for that

**Comment by t-oster** [comment]
Because Chats is forbidden by many providers (gmail,yahoo)

**Comment by t-oster** [comment]
I'm not sure about Chat though (without s) maybe this is allowed

**Comment by STPKITT** [comment]
If the decision will be EmailChats, which I'd vote for, it should be written E-Mail-Chats or at least Email-Chats though.

**Comment by Hocceruser** [comment]
@t-oster 
> I'm not sure about Chat though (without s) maybe this is allowed

r10s wrote:
> * it should not be reserved, so. eg. `Chat` may be a bad choice as well

@STPKITT 
> If the decision will be EmailChats, which I'd vote for, it should be written E-Mail-Chats or at least Email-Chats though.

IMHO using no `-` is better as I think that chances are good that there is at least one provider or whatever having problems with these characters.

**Comment by r10s** [comment]
just tried to create the _label_ `Chat` in gmail, it does not work.

**Comment by hpk42** [comment]
I am skeptical of the implied generality of the EmailChats claim.  On a more abstract level (sic!), i grew more careful of "over-abstraction" which leads to un-needed complexity.  If two projects want to interoperate it often is better and easier to just pragmatically achieve it  rather than to create and specify a supposedly abstract base to which both projects map. 

FWIW the autocrypt  effort is an actual specification effort that is implemented independently in multiple MUAs and many folks contributed to writing the spec.  By contrast, Delta's handling of chat messeges contains a lot of ad-hoc decisions by few (sometimes one) person.  There are some early Delta spec efforts but compared to autocrypt they lack detail, and i guess they could not by themselves be used to  interoperate with Delta.  I find the "standards" notion is not real and not substantial enough yet to warrant the mismatch between the app and the folder name.  In my conversations i still encounter a lot of confusion about Deltachat -- it's for many still an unusual, new effort.   Claiming something unspecified like "EmailChat" when there is no real EmailChat community (i.e. people from other than DeltaChat projects) adds to confusion.  Mind you, I do like the idea of having an EmailChat specification and would try to help such an effort when the time comes (maybe in 2019?).
    
TLDR; My preference is to go for "DeltaChat" (and INBOX.Deltachat) now and aim for "EmailChat" later.  Given #173 think it's not hard to do this rename robustly when the time comes. which i hope it will.

**Comment by t-oster** [comment]
IMHO we should not take too much time to think about this, because right now this is a show-stopper for many popular E-Mail providers. `DeltaChat` is fine with me. I could also live with a configurable name and since this does not affect communication with other users, only synchronization between multiple "Email-Chat" compliant clients on the same Mailbox, I don't think it is that important for the Standard.
I think the core developers should just make a decision in order to move forward.

**Comment by r10s** [comment]
i agree with @t-oster and @hpk42 above. looking forward to the `DeltaChat` folder now :)

**Comment by csb0730** [comment]
I agree too
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by t-oster** [comment]
If you want i can start working on a PR as soon as #175 is merged.

**Comment by testbird** [comment]
Sorry, I missed most currently, and only stopping by, and would like to not remind you about the libpurble and telepathy plugins https://gitlab.com/lupine/purple-plugin-delta/issues/5 which are using deltachat-core.

It's not nice to users changing the name later and looks kind of awkward for a public "core" library to use an app specific app name, instad of something simple and generic like `Chats`, or --as that introduced errors--  a correction to `EmailChats` or similar.

It'a simple measure not to put obstacles in your own way.

**Comment by r10s** [comment]
For libpurple the name should be fine, in fact, it is also named as `purple-plugin-delta`.

**Comment by gerroon** [comment]
Would it be better to let the user to define the name optonally ?

**Comment by STPKITT** [comment]
A default is required obviously, but in my opinion it should definitely be configurable.

**Comment by gerroon** [comment]
@STPKITT 
Yeah a default is fine naturally but a custom name would make it more fitting to people's needs,

**Comment by violoncelloch** [comment]
But a custom name would require the users to configure the same name in every delta chat installation :/

**Comment by gerroon** [comment]
@violoncelloch 

Sounds proper! That is exactly I want to do as a user, not a hinderance. If you use the default nothing to worry about.

**Comment by STPKITT** [comment]
@violoncelloch True but I agree with @gerroon plus the problem could be eased by transfering that setting alongside the export/import backup function.
